src/compiler/utilities.ts

```diff
@@ -850,10 +850,88 @@ namespace ts {
         return node && (node.kind === SyntaxKind.GetAccessor || node.kind === SyntaxKind.SetAccessor);
     }
 
-    export function isClassLike(node: Node): node is ClassLikeDeclaration {
+    export function isClassLike(node: Node | undefined): node is ClassLikeDeclaration {

```

Comment:
I would rather see type predicates like this take defined inputs. I have a PR (#16121) making this public and would prefer to keep just `Node` as the parameter type; we can keep the `node &&` in the body for now, but if we ever start using `--strictNullChecks` it should become the caller's responsibility.